internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1000,17 +1002,20 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
 			Replicas:    &replicas,
 			Selector: metav1.LabelSelector{
 				MatchLabels: map[string]string{
-					"machine-set-matching-label": "true",
+					"preserved-label": "preserved-value",
 				},
 			},
 			Template: clusterv1.MachineTemplateSpec{
 				ObjectMeta: clusterv1.ObjectMeta{
 					Labels: map[string]string{
```

Comment:
nit: maybe just add a godoc comments here that those labels/annotations are preserved, dropped, modified later when we update a MachineSet to run an in-place mutation on Machine, BootstrapConfig and InfraMachine

(Independent of that, definitely like the new labels / annotations a lot more than what we had before)